app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        if isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Union[Distance, int]) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Union[Distance, int]) -> Distance:
+        return Distance(round(self.km / other, 2))
```

Comment:
The `__truediv__` method should also handle division by another `Distance` object. Implementing this logic will make your class more robust and versatile.